crates/core/app/src/server/mempool.rs

```diff
@@ -12,21 +12,10 @@ use tracing::Instrument;
 
 use crate::{app::App, metrics};
 
-/// When using ABCI, we can't control block proposal directly, so we could
-/// potentially end up creating blocks with mutually incompatible transactions.
-/// While we'd reject one of them during execution, it's nicer to try to filter
-/// them out at the mempool stage. Currently, the way we do this is by having
-/// the mempool worker maintain an ephemeral fork of the entire execution state,
-/// and execute incoming transactions against the fork.  This prevents
-/// conflicting transactions in the local mempool, since we'll update the fork,
-/// then reject the second transaction against the forked state. When we learn a
-/// new state has been committed, we discard and recreate the ephemeral fork.
-///
-/// After switching to ABCI++, we can eliminate this mechanism and just build
-/// blocks we want.
+/// A mempool service that applies transaction checks against an isolated application fork.
 pub struct Mempool {
     queue: mpsc::Receiver<Message<Request, Response, tower::BoxError>>,
-    app: App,
+    snapshot: Snapshot,
```

Comment:
Why do we want the mempool to hold on to a `snapshot`? Previously, it was stateful and then it held a forked `App` state that was periodically reset, but do we need any of that now? What if the mempool just has a `Storage` handle and pulls the latest snapshot on every request?